moabb/evaluations/evaluations.py

```diff
@@ -177,142 +181,108 @@ def _evaluate(
                 for subject in tqdm(
                     dataset.subject_list, desc=f"{dataset.code}-WithinSession"
                 )
+                for pipeline in self.results.not_yet_computed(
+                    pipelines, dataset, subject, process_pipeline
+                )
             )
 
         # Concatenate the results from all subjects
         yield from [res for subject_results in results for res in subject_results]
 
+    @memory.cache
     def _evaluate_subject(
         self,
         dataset,
-        pipelines,
-        param_grid,
+        pipeline,
         subject,
-        process_pipeline,
         postprocess_pipeline,
     ):
-        # Progress Bar at subject level
-        # check if we already have result for this subject/pipeline
-        # we might need a better granularity, if we query the DB
-        run_pipes = self.results.not_yet_computed(
-            pipelines, dataset, subject, process_pipeline
-        )
-        if len(run_pipes) == 0:
-            return []
-
-        # get the data
-        X, y, metadata = self.paradigm.get_data(
+        # Getting the data
+        X, labels, metadata = self.paradigm.get_data(
             dataset=dataset,
             subjects=[subject],
             return_epochs=self.return_epochs,
             return_raws=self.return_raws,
             postprocess_pipeline=postprocess_pipeline,
         )
         subject_results = []
-        # iterate over sessions
+
+        y = []
+        # **WRONG**
+        # We need to think in a better way to do this...
+        # Before: For each session, we perform a Label Encoder
         for session in np.unique(metadata.session):
             ix = metadata.session == session
+            y.append(self.encode_labels(labels[ix], self.mne_labels))
 
-            for name, clf in run_pipes.items():
-                if _carbonfootprint:
-                    # Initialize CodeCarbon
-                    tracker = EmissionsTracker(save_to_file=False, log_level="error")
-                    tracker.start()
-                t_start = time()
-                cv = StratifiedKFold(5, shuffle=True, random_state=self.random_state)
-                inner_cv = StratifiedKFold(
-                    3, shuffle=True, random_state=self.random_state
-                )
-                scorer = get_scorer(self.paradigm.scoring)
-                le = LabelEncoder()
-                y_cv = le.fit_transform(y[ix])
-                X_ = X[ix]
-                y_ = y[ix] if self.mne_labels else y_cv
+        name, clf = pipeline.items()
+        if _carbonfootprint:
+            # Initialize CodeCarbon
+            tracker = EmissionsTracker(save_to_file=False, log_level="error")
+            tracker.start()
+        t_start = time()
 
-                grid_clf = clone(clf)
+        # To-do: find a way to expose this for.
```

Comment:
Yes, I think that if we want to only have one `Parallel` to avoid nesting them, we should put it here instead of across the datasets and subjects for two reasons:
- parallel calls between subjects and datasets means loading a lot of data in parallel so it's not super efficient
- plus, if the user wants to also do parallel calls between datasets or subjects, they can launch multiple scripts, each with a different subject.